src/components/BlogForm/AddBlog.tsx

```diff
@@ -0,0 +1,178 @@
+import { useSnackbar } from "notistack";
+import React, { useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import ReactQuill from "react-quill";
+import "react-quill/dist/quill.bubble.css";
+import "react-quill/dist/quill.snow.css";
+import { addFormData } from "./types";
+import { modules } from "./modulesRQ";
+
+const tagOptions = ["Finance", "Jobs", "Education"];
+
+const AddBlog = () => {
+  const { data: session } = useSession();
+  const { enqueueSnackbar } = useSnackbar();
+  const [content, setContent] = useState("");
+  const [heroPhoto, setHeroPhoto] = useState<File | null>(null);
+  const { control, handleSubmit, reset } = useForm({
+    defaultValues: {
+      title: "",
+      summary: "",
+      tags: "",
+    },
+  });
+
+  const handleImageChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.target;
+    if (files && files.length > 0) {
+      setHeroPhoto(files[0]);
+      enqueueSnackbar("Image Selected");
+    } else {
+      enqueueSnackbar("No file selected.", { variant: "error" });
+    }
+  };
+
+  if (!session?.user) {
+    return <LoadingScreen />;
+  }
+  const { user } = session;
+
+  const onSubmit = (data: addFormData) => {
+    if (!heroPhoto) {
+      return enqueueSnackbar("No here image selected.", { variant: "error" });
+    }
+    let imageId;
+    let blogSlug;
+
+    blogStorage(heroPhoto, data.title).then((res) => {
+      ({ imageId, blogSlug } = res);
+      postBlog(blogSlug, {
+        ...data,
+        author: user.name,
+        imageId: imageId,
+        slug: blogSlug,
+        content: content,
+      })
+        .then(() => enqueueSnackbar("Blog card has been added"))
+        .catch((e) =>
+          enqueueSnackbar(`Error adding the blog card: ${e.message}`, {
+            variant: "error",
+          })
+        )
+        .finally(() => {
+          reset();
+          setHeroPhoto(null);
+          setContent("");
+        });
+    });
+  };
+
+  return (
+    <Container maxWidth="md">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Stack direction={"column"} spacing={2} py={5}>
+          <Controller
+            name="title"
+            control={control}
+            rules={{ required: true }}
+            render={({
+              field: { onChange, value },
+              fieldState: { error },
+            }: any) => (
+              <TextField
+                onChange={onChange}
+                value={value}
+                label={"Blog Title"}
+                error={!!error}
+                helperText={!!error && "A title is required"}
+                size="small"
+              />
+            )}
+          />
+          <Controller
+            name="tags"
+            control={control}
+            rules={{ required: true }}
+            render={({
+              field: { onChange, value },
+              fieldState: { error },
+            }: any) => (
+              <TextField
```

Comment:
use an Autocomplete input instead of text field

Reply:
why?

Reply:
because you're rendering multiple options inside the TextField which is not what its used for. instead, just create a list of options and provide into Autocomplete as a prop. Autocomplete allows a user to search for the options which is useful if there are lots of them